cmd/controller-manager/app/options/options.go

```diff
@@ -106,6 +121,19 @@ func (s *ServerOption) AddFlags(fs *pflag.FlagSet) {
 	fs.BoolVar(&s.InheritOwnerAnnotations, "inherit-owner-annotations", true, "Enable inherit owner annotations for pods when create podgroup; it is enabled by default")
 	fs.Uint32Var(&s.WorkerThreadsForPG, "worker-threads-for-podgroup", defaultPodGroupWorkers, "The number of threads syncing podgroup operations. The larger the number, the faster the podgroup processing, but requires more CPU load.")
 	fs.Uint32Var(&s.WorkerThreadsForGC, "worker-threads-for-gc", defaultGCWorkers, "The number of threads for recycling jobs. The larger the number, the faster the job recycling, but requires more CPU load.")
+	fs.StringSliceVar(&s.Controllers, "controllers", KnownControllers(), "Specify controller gates. Use '*' for all controllers, \"+gc-controller,+job-controller,+jobflow-controller,+jobtemplate-controller,+pg-controller,+queue-controller\" to enable specific controllers, "+
+		"and \"-gc-controller,-job-controller,-jobflow-controller,-jobtemplate-controller,-pg-controller,-queue-controller\" to disable specific controllers.")
+}
+
+// KnownControllers returns a list of known controllers.
+func KnownControllers() []string {
```

Comment:
Maybe move the func to `cmd/controller-manager/main.go` is better, because it's where the controllers are registered, and we should remove importing controllers in `cmd/controller-manager/app/options/options.go`.

Reply:
Hi, please also notice this piece, we can merge it after this is done: ）

Reply:
I will deal with this issue in the next two days. Thanks for the reminder.

Reply:
In main.go, I use the knownControllers function variable to ensure the simplicity of main.go
```go
// knownControllers is a list of all known controllers.
	var knownControllers = func() []string {
		controllerNames := []string{}
		fn := func(controller framework.Controller) {
			controllerNames = append(controllerNames, controller.Name())
		}
		framework.ForeachController(fn)
		sort.Strings(controllerNames)
		return controllerNames
	}

```